internal/hook/pod_backup_post_hook_handler.go

```diff
@@ -0,0 +1,58 @@
+package hook
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	velerov1 "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
+	"github.com/vmware-tanzu/velero/pkg/podexec"
+	"github.com/vmware-tanzu/velero/pkg/podvolume"
+)
+
+var _ ResourceHookHandler = &podBackupPostHookHandler{}
+
+func NewPodBackupPostHookHandler(backupper podvolume.Backupper, executor podexec.PodCommandExecutor) ResourceHookHandler {
+	return &podBackupPostHookHandler{
+		podVolumeBackupper: backupper,
+		podCommandExecutor: executor,
+	}
+}
+
+// podBackupPostHookHandler handles the post backup hooks for pods
+type podBackupPostHookHandler struct {
+	podVolumeBackupper podvolume.Backupper
+	podCommandExecutor podexec.PodCommandExecutor
+}
+
+// Don't need to wait when using CSI snapshot, because:
+//
+//	PVCs are backed up before Pods (Pod's BIA returns PVCs as additional resources).
+//	PVC's BIA creates VS and returns it as additional resource.
+//	VS's BIA waits VSC's handle ready: https://github.com/vmware-tanzu/velero/blob/v1.14.1/pkg/backup/actions/csi/volumesnapshot_action.go#L118
+//
+// So when Pods are backed up (when hooks execute), the snapshot is taken already
+func (p *podBackupPostHookHandler) WaitUntilReadyToExec(ctx context.Context, log logrus.FieldLogger, res *unstructured.Unstructured) error {
+	// wait all PVBs processed
+	_, err := p.podVolumeBackupper.WaitAllPodVolumesProcessed(log)
```

Comment:
Here is a limitation:
The logic here waits all PVBs of the whole backup to be completed before executing the hooks. This delays the hook execution of one pod although it isn't necessary.

The ideal behavior should be only wait the PVBs belongs to one ItemBlock to be completed before executing the hooks, but this involves the refactor for podvolume.Backupper to make is capable to check whether the PVBs of one ItemBlock completed. The effort isn't tiny, so created another issue #8506 to track.